### DISCUSSION TITLE
Add more  objectName on qml Item

### DISCUSSION THREAD

**Discussion by qsavoye** [opengisch/QField#5674] [Ideas]
To make development of qfield plugin easier it would be nice to add more objectName on qml elements. Indeed it is easier and more sustainable for further update to use `findItemByObjectName` function to search qml element than search by index (for example : `iface.mainWindow().contentItem.children[4].children[9]`)

**Comment by nirvn** [comment]
@qsavoye , which items are you interested in?

**Reply by qsavoye** [reply]
For my plugin https://github.com/qsavoye/QField-Geocoder-Addresses-Plugin, I need the following objectName to avoid search by indices,

- [ ] overlayFeatureFormDrawer
- [ ] locatorItem
- [ ] searchFieldRect
- [ ] searchField
- [ ] resultsBox

Maybe some people would add more item to the list.

Would you like me to create a pull request with these changes ?

**Reply by nirvn** [reply]
@qsavoye , you can do so for overlayFeatureFormDrawer and locatorItem.

For the other ones they are too generic and prone to change.

That said, we'll come up with a proper mechanism for plugins to register themselves with the search bar, you might want to wait for that.
